assets/src/edit-story/components/checklist/priorityChecks.js

```diff
@@ -27,8 +28,23 @@ import StoryPosterPortraitSize from './checks/storyPosterPortraitSize';
 import StoryTitleLength from './checks/storyTitleLength';
 import VideoElementMissingPoster from './checks/videoElementMissingPoster';
 import { ChecklistCategoryProvider } from './checkCountContext';
+import VideoOptimization from './checks/videoOptimization';
 
 export function PriorityChecks() {
+  const { currentUser } = useCurrentUser(({ state, actions }) => ({
+    currentUser: state.currentUser,
+    toggleWebStoriesMediaOptimization:
+      actions.toggleWebStoriesMediaOptimization,
+  }));
+  const {
+    capabilities: { hasUploadMediaAction },
+  } = useConfig();
+
+  const isVideoOptimizationSettingEnabled =
+    Boolean(window?.crossOriginIsolated) &&
+    currentUser?.meta?.web_stories_media_optimization &&
+    hasUploadMediaAction;
```

Comment:
Please do not reinvent the wheel here. Just use the state from the ffmpeg hook. 

```
  const {
    isFeatureEnabled,
    isTranscodingEnabled,
  } = useFFmpeg();
```

So this becomes 
```
  const isVideoOptimizationSettingEnabled =
    hasUploadMediaAction && isFeatureEnabled && isTranscodingEnabled;
```